ext/mysqlnd/mysqlnd_ps_codec.c

```diff
@@ -241,7 +241,11 @@ ps_fetch_time(zval * zv, const MYSQLND_FIELD * const field, const unsigned int p
 		t.time_type = MYSQLND_TIMESTAMP_TIME;
 	}
 
-	length = mnd_sprintf(&value, 0, "%s%02u:%02u:%02u", (t.neg ? "-" : ""), t.hour, t.minute, t.second);
+    if (field->decimals > 0) {
+        length = mnd_sprintf(&value, 0, "%s%02u:%02u:%02u.%0*u", (t.neg ? "-" : ""), t.hour, t.minute, t.second, field->decimals, t.second_part);
```

Comment:
The `second_part` member is `zend_ulong`. It would be cleaner to use ZEND_ULONG_FMT_SPEC instead of `u`. Please see `zend_long.h`.

Reply:
Of course, skipping that could cause trouble with e.g. future 16bit platforms.

Reply:
After checking your recommendations in practice, I am fairly sure that using ZEND_ULONG_FMT_SPEC would *not* work.
What is needed here is to format the same ulong as an unsigned zero-padded decimal having any number of digits between 1-6.
While ZEND_ULONG_FMT_SPEC resolves - at least here - to PRIu64, we do not want an unsigned long here but a predetermined amount of digits, zero padded.

Reply:
`zend_ulong` is dynamic, it's 64- or 32-bit depending on the platform used. Using the macro is trivial. 16-bit platforms are not supported at all, so nothing to care. Principally, one could also leave `u` as the fractional part won't be that big anyway. It's just a formal thing - the format to match the datatype. It doesn't affect the field width, as the decimal places number is still being passed. I'm going to check once more tomorrow.

Thanks.

Reply:
Okay, after recompiling PHP with --enable-debug, and using mysql_debug, and adding some extra DBG_INF_FMT macros, it became apparent that while the decimals member is **always right** (2, 4, 6, the precision that is configured), the second_part member has its value **always set to microseconds** internally in mysql.

Sigh.

As such, I think we are not heading towards the long format, since we want to get "77" from "770000" with a decimals value of 2, and even "9" from a long value of "900000" with a decimals value of 1.

Reply:
This also means the **"second_part"** member name is a misnomer, since it **is always microseconds**, no matter the actual precision configured.